### DISCUSSION TITLE
Руководство по настройкам

### DISCUSSION THREAD

**Discussion by dosinabox** [dosinabox/g2nr_unofficial_update#352] [General]
- Включить начало четвертой главы в Хоринисе
- Включить полное удаление NPC из мира
- Включить альтернативное обучение у Беннета
- Включить повышенный уровень сложности
- Включить алхимию из Готики 2 без аддона
- Отделить шлемы от доспехов
- Включить честный расчет стоимости обучения
- :negative_squared_cross_mark:Выключить усиленных предводителей орков
- Включить возможность прохождения без аддона
- Включить возможность прохождения без гильдии
- Включить выкуп гроссбуха у Торбена и Корагона
- Включить лимит запаса стрел и болтов у торговцев
- Включить бесконечную сталь и мензурки у торговцев
- Включить режим трусости у людей и животных
- Включить игнорирование бонусов при прокачке
- Включить рандомные гоблинские ягоды у паладинов
- Включить выбивание бесконечных яблок
- Включить фиксированный опыт по главам
- :negative_squared_cross_mark:Выключить опыт с бесконечно призываемых скелетов
- Включить показ заголовков при обновлении дневника

**Comment by dosinabox** [comment]
Добавлена опция "Получить книгу статистики".
Возможно, для настроек нужен новый отдельный помощник (?).

**Comment by Kor-Angar** [comment]
> Возможно, для настроек нужен новый отдельный помощник (?).

я думаю настройки пора вшивать в диалог с Ксардасом с возможностью его скрыть.

**Comment by Kor-Angar** [comment]
> Включить фиксированный опыт по главам

вот это тебе надо расписать. я так и не понял в чем смысл этого был.

> Включить повышенный уровень сложности

это тоже

> Включить альтернативное обучение у Беннета

это правка Диего с ранним доступом к самокованным мечам из руды у Наёмника?

**Comment by dosinabox** [comment]
> я думаю настройки пора вшивать в диалог с Ксардасом с возможностью его скрыть.

Ксардаса нет в Яркендаре, Ирдорате и Минентале, а из Хориниса он пропадает в пятой главе. Плохой выбор, потому что доступен не всегда. Можно привязать к алтарям Белиара, они есть во всех мирах. 
А как скрывать?

> вот это тебе надо расписать. я так и не понял в чем смысл этого был.

Полный отказ от динамического XP_Ambient и возможность выставить столько, сколько хочется фиксировано (от 1 до 6 главы).

> это тоже

Возможность снизить весь получаемый опыт (от -10% до -50%).

> это правка Диего с ранним доступом к самокованным мечам из руды у Наёмника?

Правка Диего, но подробности не помню.

**Comment by Kor-Angar** [comment]
> Ксардаса нет в Яркендаре, Ирдорате и Минентале, а из Хориниса он пропадает в пятой главе. Плохой выбор, потому что доступен не всегда. Можно привязать к алтарям Белиара, они есть во всех мирах.

хороший выбор, так как он идёт сразу и не заметить это невозможно. так же можно будет вызвать через консоль помощника и настроить. тем более что настраивать игру надо в первую очередь в начале.

> А как скрывать?

в диалоге у Ксарсада в настройках "скрыть настройки" и этот диалог пропадает.

> Полный отказ от динамического XP_Ambient и возможность выставить столько, сколько хочется фиксировано (от 1 до 6 главы).

это я знаю. ты так напиши что это такое. теперь в первой главе за разговор ни о чём вы сможете получать по 300 опыта, таким образом в первой главе вы сможете поднять 25 уровней. решает такую то задачу / проблема. полностью ломает баланс игры.

> Возможность снизить весь получаемый опыт (от -10% до -50%).

такой же принцип как и выше. снижает получаемый опыт, полностью ломает баланс игры. информации как оно  нет, но по ощущениям вы сможете к концу 5 главы набрать 25 уровней. такое вот решение из области Правительства вы просили как то улучшить жизнь, вот мы как то её улучшили. 

> Правка Диего, но подробности не помню.

из той же области. это опция делает доступными самокованные мечи на ранней стадии игры. ломаем баланс игры без использования чит кодов и марвина.

***
т.е. я не смогу это объективно описать. про другие позиции могу, про это не смогу :-)

**Comment by Kor-Angar** [comment]
добавь в первый пост, что по дефауту включено, а что выключено.

**Comment by dosinabox** [comment]
Если написано "включить", то по умолчанию выключено и наоборот

**Comment by Kor-Angar** [comment]
может разные слова использовать у помощника что бы оно как то отличалось визуально? :-)

> добавь в первый пост

вроде отметил

**Comment by Kor-Angar** [comment]
надо попробовать разбить их на какие подкатегории. например:

**Сюжет**
Включить возможность прохождения без аддона
Включить возможность прохождения без гильдии

**Баланс**
Включить повышенный уровень сложности
Включить честный расчет стоимости обучения
Включить альтернативное обучение у Беннета
Выключить усиленных предводителей орков
Выключить опыт с бесконечно призываемых скелетов
Включить рандомные гоблинские ягоды у паладинов
Включить фиксированный опыт по главам
Включить игнорирование бонусов при прокачке

**Лут**?
Включить лимит запаса стрел и болтов у торговцев
Включить бесконечную сталь и мензурки у торговцев
Включить выбивание бесконечных яблок

**Визуализация** 
Отделить шлемы от доспехов
Включить режим трусости у людей и животных

**разное**
Включить начало четвертой главы в Хоринисе
Включить полное удаление NPC из мира
Включить алхимию из Готики 2 без аддона
Включить выкуп гроссбуха у Торбена и Корагона
Включить показ заголовков при обновлении дневника

**Comment by dosinabox** [comment]
Про новые возможности помощника по персонажу тоже следует написать. Смена внешности и походки, прокачка регенерации маны и здоровья.

**Comment by Kor-Angar** [comment]
подскажи пожалуйста сколько ушло бы ЛП а прокачку силы если играть с  включенной опцией "игнорирование бонусов при прокачке"


lvl |   |   |   | ЛП
-- | -- | -- | -- | --
2 | сила +20 | 20 | Ларес | 20
13 | сила +3 | 3 | Мортис | 6
14 | сила +1 | 1 | Мортис | 2
16 | сила +5 | 5 | Диего | 10
17 | сила +5 | 5 | Диего | 10
18 | сила +3 | 3 | Диего | 6
19 | Сила +2 | 2 | Мортис | 6
20 | Сила +4 | 4 | Мортис | 12
22 | Сила +5 | 5 | Мортис | 15
24 | Сила +5 | 5 | Мортис | 15
25 | Сила +4 | 4 | Бандит | 12
27 | Сила +5 | 5 | Мортис | 20
31 | Сила +2 | 2 | Ингмар | 10
34 | Сила +5 | 5 |   | 25
35 | Сила +3 | 3 |   | 15
  |   | 72 |   | 184

т.е. условно за ЛП сила прокачена **на** 72, т.е. +20 +30 +22. таким образом 20х1 + 30х2 + 22х3 = 20+60+66 = 80+66=146. так?

**Comment by Kor-Angar** [comment]
точнее наверное даже так? 24+30+18,  24х1+30х2+18х3=24+60+54=60+78=138
или я неправильно понимаю как работает "игнорирование бонусов при прокачке" ?

**Comment by Kor-Angar** [comment]
может имеет смысл часть настроек вынести в ini файл и считывать/записывать так же там?  в zparser есть функции которые это делают. я пробовал на языках там всё достаточно просто. могу попробовать и на неоф. обновление с настройками.

возможно кому то будет удобнее привычнее делать это там.

**Reply by Kor-Angar** [reply]
```D
// Считывает целочисленное значение параметра из ini
// файла optName ("Gothic", "Mod", "SystemPack"), секции section,
// вхождения entry, где значением по умолчанию считыется default
func int Hlp_ReadOptionInt(var string optName, var string section, var string entry, var int default)

// Считывает вещественное значение параметра из ini
// файла optName ("Gothic", "Mod", "SystemPack"), секции section,
// вхождения entry, где значением по умолчанию считыется default
func float Hlp_ReadOptionFloat(var string optName, var string section, var string entry, var float default)

// Считывает строковое значение параметра из ini
// файла optName ("Gothic", "Mod", "SystemPack"), секции section,
// вхождения entry, где значением по умолчанию считыется default
func string Hlp_ReadOptionString(var string optName, var string section, var string entry, var string default)

// Определяет существует ли в ini
// файле optName ("Gothic", "Mod", "SystemPack") секция section
// (при entry = "") или вхождение entry (тогда entry != "")
func int Hlp_OptionIsExists(var string optName, var string section, var string entry)

// Записывает целочисленное значение value в ini
// файл optName ("Gothic", "Mod", "SystemPack")
// в секцию section, вхождение entry
func void Hlp_WriteOptionInt(var string optName, var string section, var string entry, var int value)
```

**Reply by Kor-Angar** [reply]
то есть фактически это использование при инициализации мира `Hlp_ReadOptionInt`

и в помощнике по настройкам `Hlp_WriteOptionInt`

**Reply by Kor-Angar** [reply]
пока так.
[draft.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/10471107/draft.zip)

**Reply by Kor-Angar** [reply]
это было  лишнее MIS_ReadyforChapter4 :-)
